tests/scan/test_basic.py

```diff
@@ -240,6 +241,54 @@ def scan_nodes_from_fct(fct):
 
 
 class TestScan:
+    @pytest.mark.parametrize(
+        "rng_type",
+        [
+            np.random.default_rng,
+            np.random.RandomState,
+        ],
+    )
+    def test_inner_graph_cloning(self, rng_type):
+        r"""Scan should remove the updates-providing special properties on `RandomType`\s."""
+
+        inner_inner_rng = shared(rng_type(), name="inner_inner_rng")
+
+        y = shared(np.array(1.0, dtype=config.floatX), name="y")
+        y.default_update = y + 1
+
+        z_rng = shared(rng_type(), name="z_rng")
+        z = normal(0, 1, rng=z_rng, name="z")
+
+        z_rng_update = z.owner.outputs[0]
+        z_rng_update.name = "z_rng_update"
+        z_rng.default_update = z_rng_update
+
+        inner_rng = None
+
+        def inner_fn(x):
+            inner_rng = shared(rng_type(), name="inner_rng")
+            inner_rng.default_update = inner_inner_rng
+            inner_inner_rng.default_update = inner_rng
+
+            r = normal(x, rng=inner_rng)
+            return r + y + z, z
```

Comment:
Perhaps return the `y` as well? It should behave like `z` I assume

Reply:
There's no need to do that; it won't change the way `y` interacts with the functionality being tested.  The only reason that `z` is returned is to confirm that its computed values are all the same.

Reply:
Right but the same repeated values should (and seem to be) observed with `y`, no? 

Otherwise, is there any reason for including it in the test at all?

Reply:
> Right, but the same repeated values should (and seem to be) observed with `y`, no?
> 
> Otherwise, is there any reason for including it in the test at all?

Unlike `z`, `y`'s update is directly attached to itself and not to the shared variable of an input, so it serves as the simplest inner-graph shared-variable-with-an-outer-scoped-update case.

Aside from that, it doesn't differ from `z_rng` in terms of how its treated (wrt. the relevant updates-based functionality) during compilation and evaluation, so we don't need to check the computed output as well; we only need to check that the `default_update` is still present, as we do.

For that matter, we could simply end the test after the `hasattr` checks, but the full `function` compilation at least confirms that the original error is gone.